src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,143 @@
+package frc.robot.leds;
+
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+import java.awt.*;
+
+public class LedStrip extends SubsystemBase {
+    private final AddressableLED ledStrip;
+    private AddressableLEDBuffer ledBuffer;
+
+    private double blinkTime = 1;
+    private double fadeDuration = 1;
+    private int rainbowHue = 0;
+
+    private Color primary = new Color(0);
+    private Color secondary = new Color(0);
+    private Color currentColor = primary;
+
+    private final Timer timer = new Timer();
+    private Translation3d solution = new Translation3d();
+    private LedMode mode = LedMode.SOLID;
+
+    public LedStrip(int port, int length) {
+        ledStrip = new AddressableLED(port);
+        ledBuffer = new AddressableLEDBuffer(length);
+
+        ledStrip.setLength(length);
+        ledStrip.setData(ledBuffer);
+        ledStrip.start();
+
+        timer.start();
+        timer.reset();
+    }
+
+    public void setLength(int length) {
+        ledBuffer = new AddressableLEDBuffer(length);
+        ledStrip.setLength(length);
+    }
+
+    public void setState(LedState state){
```

Comment:
I think it'll be more concise to store the `LedState` object directly instead of its members.